### PR TITLE
refactor: extend inline testing

### PR DESCRIPTION
**Pull request by idoros**
This PR extends `testInlineExpects` core test utility to handle more cases.

## Tasks:
- [x] change `testInlineExpects` internal implementation to be easily extended
    - change old `@check` expectation to proxy to new `@rule` / `@atrule`
    - add missing tests for unsupported nodes
- add `@analyze` and `@transform` diagnostics expectation
  - [x] test severity and message with optional label
  - [x] test location
  - [x] test word
  - [x] check against `rawAst` for locations that are removed in `outputAST`
- [x] single declaration expectation - `@decl`
- [x] support for mixed-in atrules
- [x] add changes/additions to README

## to backlog
- single test function for all tests
  - accept a single stylesheet as entry to transform
  - accept multiple entries (options for transform list)
  - inline test all entries
  - expose `stylable`, `fs`, and stylable meta and exports of stylesheets

### CODE REVIEW

**Comment by tomrav** [comment]
Looks good, left a couple of remarks.

**Review by tomrav — COMMENTED** [review]
Looks good, left a couple of comments.

File: packages/core-test-kit/test/inline-expectations.spec.ts [hunk 1]
```diff
@@ -40,256 +43,1063 @@ describe('inline-expectations', () => {
...
+            );
+        });
+        it(`should check against rawAst (for nodes that are removed in transform)`, () => {
+            const result = generateStylableResult({
+                entry: `/style.st.css`,
+                files: {
+                    '/style.st.css': {
+                        namespace: 'entry',
+                        content: `
+                            /* @transform-warn word(./x.st.css) ${STImport.diagnostics.UNKNOWN_IMPORTED_FILE(
```
[the remark this answers is not in the document]
Comment (idoros) [reply packages/core-test-kit/test/inline-expectations.spec.ts:1039]: added labels for the expectations to better indicate their purpose (`should match` / `should fail`) and added a comment before the throw expectation explaining the single error expectation for emphasis.

File: packages/core-test-kit/src/inline-expectation.ts [hunk 2]
```diff
@@ -47,161 +59,440 @@ interface AtRuleCheck {
...
+            expectedNotFound: testInlineExpectsErrors.diagnosticExpectedNotFound,
+        }
+    );
+    if (error) {
+        result.errors.push(error);
+    }
+    return result;
+}
+function transformTest(
+    { meta }: Context,
```
[the remark this answers is not in the document]
Comment (idoros) [reply packages/core-test-kit/src/inline-expectation.ts:310]: done. also found an unused error while unify the `@analyze/@transform` commons.

**Pull request merged**